src/vs/vscode.proposed.d.ts

```diff
@@ -2276,20 +2276,58 @@ declare module 'vscode' {
 
 	//#region https://github.com/Microsoft/vscode/issues/15178
 
-	// TODO@API must be a class
-	// TODO@API @lramos15 Call this XYZTabs
-	export interface OpenEditorInfo {
-		//viewColumn: ViewColumn; // todo @lramos15
-		name: string;
-		resource: Uri; // make optional @lramos15
+	/**
+	 * Represents a tab within the window
+	 */
+	export interface Tab {
+		/**
+		 * The text displayed on the tab
+		 */
+		label: string;
+
+		/**
+		 * The position of the tab
+		 */
+		viewColumn: ViewColumn;
+
+		/**
+		 * The resource represented by the tab.
+		 * May be undefined in certain cases such as diff editor and marketplace tabs
```

Comment:
I think I would leave out "marketplace tabs" as people probably don't know what that is. I would give other examples like terminal.

Reply:
👍 Yeah, stay generic like "The resource represnted by the tab if available". And then add a blurb that not all editor types are associated with a resource